org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AbstractTryWithResourcesMatcher.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.analysis.filter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.VarInsnNode;
+
+abstract class AbstractTryWithResourcesMatcher {
```

Comment:
@marchof indeed and I've been thinking about this - in such case `SynchronizedFilter` will pay cost of unnecessary allocation of `Map vars`. This cost can be removed by addition of a contract for subclasses - it should be initialized in subclasses if used. Before doing either of those changes I want to do at least some basic measurement as a sanity check.